forte/trainer/base/base_trainer_new.py

```diff
@@ -0,0 +1,258 @@
+                        },
+                        "char_tag": {
+                            "type": TrainPreprocessor.DATA_INPUT,
+                            "extractor": char_extractor
+                        },
+                        "output_tag": {
+                            "type": TrainPreprocessor.DATA_OUTPUT,
+                            "extractor": output_extractor
+                        }
+                    }
+                }
+
+                return tp_request
+
+            def create_tp_config(self) -> Dict:
+                tp_config: Dict = {
+                    "dataset": {
+                        "batch_size": 512
+                    }
+                }
+
+                return tp_config
+
+            def create_pack_generator(self) -> Iterator[DataPack]:
+                reader = CoNLL03Reader()
+                train_pl: Pipeline = Pipeline()
+                train_pl.set_reader(reader)
+                train_pl.initialize()
+                pack_generator: Iterator[DataPack] = \
+                    train_pl.process_dataset(self.config_data.train_path)
+
+                return pack_generator
+
+            def train(self):
+                schemes: Dict = self.train_preprocessor.request["schemes"]
+                text_extractor: BaseExtractor = \
+                    schemes["text_tag"]["extractor"]
+                char_extractor: BaseExtractor = \
+                    schemes["char_tag"]["extractor"]
+                output_extractor: BaseExtractor = \
+                    schemes["output_tag"]["extractor"]
+
+                model: BiRecurrentConvCRF = \
+                    BiRecurrentConvCRF(word_vocab=text_extractor.get_dict(),
+                                       char_vocab_size=char_extractor.size(),
+                                       tag_vocab_size=output_extractor.size(),
+                                       config_model=self.config_model)
+                model.to(self.device)
+
+                optim: Optimizer = SGD(model.parameters())
+
+                tp = self.train_preprocessor
+
+                # Train for 10 epochs
+                for epoch in range(10):
+                    # Get iterator of preprocessed batch of train data
+                    batch_iter: Iterator[Batch] = tp.get_train_batch_iterator()
+
+                    for batch in tqdm(batch_iter):
+                        word = batch["text_tag"]["data"]
+                        char = batch["char_tag"]["data"]
+                        output = batch["output_tag"]["data"]
+                        word_masks = batch["text_tag"]["masks"][0]
+
+                        optim.zero_grad()
+
+                        loss = model(word, char, output, mask=word_masks)
+
+                        loss.backward()
+                        optim.step()
+    """
+
+    def __init__(self):
+        self._tp_request: Optional[Dict] = None
+        self._tp_config: Optional[Dict] = None
```

Comment:
For `Dict` types maybe we don't need to init as `None`, we can do empty Dict

Reply:
Fixed.